index.js

```diff
@@ -1,34 +1,53 @@
-import makeTemplate from "lodash.template";
-import { visitParents as visit } from "unist-util-visit-parents";
+/**
+ * A rehype plugin to replace template strings based on lodash.template.
+ *
+ * @since 0.0.0
+ * @package rehype-lodash-template
+ * @author Viktor Yakubiv (www.yakubiv.com)
+ * @copyright Viktor Yakubiv & Contributors
+ * @license MIT
+ */
 
-const attach = (options = {}) => {
-  const { values = {}, templateSettings } = options;
+import makeTemplate from "lodash.template";
+import { visitParents } from "unist-util-visit-parents";
 
-  const cache = new Map();
+/**
+ * A rehype plugin to replace template strings based on lodash.template.
+ *
+ * @since 0.0.0
+ * @author Viktor Yakubiv (www.yakubiv.com)
+ *
+ * @param {object} param0 { values, templateSettings }
+ * @returns Rehype Plugin
+ */
+export default function attach({ values, templateSettings }) {
+  const valueCache = new Map();
 
-  const template = (str) => {
-    if (!cache.has(str)) {
-      cache.set(str, makeTemplate(str, templateSettings));
+  const getTemplate = (identifier) => {
```

Comment:
This is supposed to be `_.template` decorated with cache. Therefore, `_.template` was renamed to `makeTemplate` and this was left to be just `template`. It's interesting, this was not clear.